cylc/flow/subprocpool.py

```diff
@@ -557,12 +557,12 @@ def _run_callback(callback, args_=None):
                 # Backup, get a platform name from the config:
                 for arg in callback_args:
                     if isinstance(arg, TaskProxy):
-                        platform_name = arg.tdef.rtconfig['platform']
+                        platform_name = arg.platform['name']
```

Comment:
Separate issue, but a confusing bit of logging output: 

```
WARNING - platform: localhost - Could not connect to xcel00.
    * xcel00 has been added to the list of unreachable hosts
    * jobs-submit will retry if another host is available.
```

Emitted at line 574 as a result of this.

Reply:
This turned out to be a 🔴 🐟 : The bad path in the logging sorted itself out as a  result of the other change. 😄